starfish/image/_filter/linear_unmixing.py

```diff
@@ -0,0 +1,107 @@
+from functools import partial
+from typing import Optional
+
+import numpy as np
+
+from starfish.imagestack.imagestack import ImageStack
+from starfish.types import Axes
+from starfish.util import click
+from ._base import FilterAlgorithmBase
+from .util import preserve_float_range
+
+
+class LinearUnmixing(FilterAlgorithmBase):
+
+    def __init__(self, coeff_mat: np.ndarray) -> None:
+        """Image scaling filter
+
+        Parameters
+        ----------
+        coeff_mat : np.ndarray
+            matrix of the linear unmixing coefficients. Should take the form:
+            B = AX, where B are the unmixed values, A is coeff_mat and X are
+            the observed values.
+
+        """
+        self.coeff_mat = coeff_mat
+
+    _DEFAULT_TESTING_PARAMETERS = {"coeff_mat": np.array([[1, -0.25], [-0.25, 1]])}
+
+    @staticmethod
+    def _unmix(image: np.ndarray, coeff_mat: np.ndarray) -> np.ndarray:
+        """Clip values of img below and above percentiles p_min and p_max
+
+        Parameters
+        ----------
+        image : np.ndarray
+            image to be scaled
+
+        coeff_mat : np.ndarray
+            matrix of the linear unmixing coefficients. Should take the form:
+            B = AX, where B are the unmixed values, A is coeff_mat and X are
+            the observed values.
+
+        Returns
+        -------
+        np.ndarray :
+          Numpy array of same shape as img
+
+        """
+        # Preallocate the new image in the same shape
+        unmixed_image = np.zeros(image.shape)
+
+        # Due to the grouping, assumes image is shape [chan, x, y]
+        n_channels = image.shape[0]
+
+        for c in range(n_channels):
```

Comment:
The framing of `B = AX` makes me think there's a one-liner that captures the logic in this for loop. Does that seem crazy? I'd be happy to talk through this on a call. 👍

Reply:
My intuition was the same. However, I couldn't figure out how to do it without the for loop. Within the for loop (i.e., for a given channel), we could do something like:
```python
np.dot(np.transpose(image[c, :, :]),  coeff_mat[c])
```
but originally, I thought my previous implementation was more clear. I'm definitely not a broadcasting wizard, so if you can figure out how to reshape the matrices to do it all in one line, I'd love to learn.